### PR TITLE
Homework14: Valeria Kudelko

### PR DESCRIPTION
**Pull request by ValeriaKudelko**

### CODE REVIEW

File: hw14task1.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+# студент на новой строке. Откройте файл и прочитайте всю
+# информацию из него. Напечатайте общее количество студентов,
+# количество студентов для каждой группы и среднюю оценку для
+# каждой группы. Допишите эту информацию в конец файла.
+# Передусмотрите возможные ошибки и обработайте их.
+
+
+def process_students_data():
+    try:
+        with open("students.txt", "w", encoding="utf-8") as file:
```
Comment (Stml89) [inline hw14task1.py]: do not hardcode filename and pass it to the function

File: hw14task2.py
Comment (Stml89) [inline hw14task2.py:1]:
```
~ 2 + 2 * 2
8.0
```
if I'm not mistaken, should be 6

**Review by Stml89 — CHANGES_REQUESTED** [review]

File: hw14task2.py
Comment (ValeriaKudelko) [reply hw14task2.py:1]: Remade

File: hw14task1.py (hunk 1 above)
Comment (ValeriaKudelko) [reply hw14task1.py]: Corrected

File: hw14task1.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+# список студентов, номер группы, их оценки. Каждый
+# студент на новой строке. Откройте файл и прочитайте всю
+# информацию из него. Напечатайте общее количество студентов,
+# количество студентов для каждой группы и среднюю оценку для
+# каждой группы. Допишите эту информацию в конец файла.
+# Передусмотрите возможные ошибки и обработайте их.
+
+
+try:
+    with open("students.txt", "w") as file:
```
Comment (Stml89) [inline hw14task1.py:17]: file name still hardcoded

File: hw14task1.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        file.write("Иван Иванов,1,9\n")
+        file.write("Евгений Евгеньев,2,8\n")
+        file.write("Олег Олегов,1,5\n")
+        file.write("Павел Павлов,2,8\n")
+        file.write("Александр Александров,3,2\n")
+        file.write("Юрий Юрьев,3,8\n")
+
+    with open("students.txt", "r") as file:
```
Comment (Stml89) [inline hw14task1.py:25]: file name still hardcoded

**Review by Stml89 — CHANGES_REQUESTED** [review]

File: hw14task1.py (hunk 2 above)
Comment (ValeriaKudelko) [reply hw14task1.py:17]: If I understand correctly, you need to make sure that the file name can be changed, to do this you need to enter a variable with the file name

Comment (Stml89) [reply hw14task1.py:17]: yes

File: hw14task1.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+# студент на новой строке. Откройте файл и прочитайте всю
+# информацию из него. Напечатайте общее количество студентов,
+# количество студентов для каждой группы и среднюю оценку для
+# каждой группы. Допишите эту информацию в конец файла.
+# Передусмотрите возможные ошибки и обработайте их.
+
+
+filename = "students.txt"
+try:
+    with open("students.txt", "w") as file:
```
Comment (Stml89) [inline hw14task1.py:17]:
```suggestion
filename = "students.txt"
try:
    with open(filename, "w") as file:
```

**Review by Stml89 — CHANGES_REQUESTED** [review]

**Pull request merged**